src/main/java/com/google/cloud/spanner/jdbc/JdbcStatement.java

```diff
@@ -105,6 +128,29 @@ public long executeLargeUpdate(String sql) throws SQLException {
     }
   }
 
+  /** Extracts the update count from the given result set and then closes the result set. */
+  private long extractUpdateCountAndClose(com.google.cloud.spanner.ResultSet resultSet)
```

Comment:
Will we be able to extract the stats without consuming the rows? IIRC, `resultSet.getStats()` can be accessed only when `resultSet.next()` returns `false` (no more rows to consume).

Reply:
Yes, I've added some additional comments to clarify. We only call this method with a `ResultSet` that has already been fully consumed. That happens in the `copyOf` method.

Reply:
Makes sense.